src/shogun/mathematics/linalg/GPUBackend.cpp

```
@@ -37,6 +37,7 @@
 #ifdef HAVE_VIENNACL
 #include <viennacl/vector.hpp>
 #include <viennacl/linalg/inner_prod.hpp>
+#include <viennacl/linalg/sum.hpp>
 #endif
```

Agreed! The creation of the class itself should fail if viennacl is not found.